src/components/Carousel/Carousel.test.tsx

```diff
@@ -0,0 +1,57 @@
+import { render, screen } from '@testing-library/react';
+import Carousel from './Carousel';
+import CarouselContent from './CarouselContent';
+import CarouselItem from './CarouselItem';
+import CarouselDots from './CarouselDots';
+import { runCommonTests } from '../../utils/testUtils';
+
+describe('Carousel', () => {
+  runCommonTests(Carousel, 'Carousel');
+
+  beforeEach(() => {
+    class MockIntersectionObserver {
+      observe = () => {}; // eslint-disable-line @typescript-eslint/no-empty-function
```

Comment:
also nice, perhaps we should move this stuff to the global test setup file so all testcases get it https://github.com/PhillipsAuctionHouse/seldon/blob/c4a909a1ac54aeffe93988c5adb83cd9a08a71cc/config/vitest/setupTest.ts#L16

Reply:
down 👍